src/shared/components/paginationControls/PaginationControls.tsx

```diff
@@ -82,9 +132,25 @@ export class TablePaginationControls extends React.Component<ITablePaginationCon
                     <Button key="prevPageBtn" disabled={!!this.props.previousPageDisabled} onClick={this.props.onPreviousPageClick}>
                         {this.props.previousButtonContent}
                     </Button>
-                    <Button key="textBetweenButtons" className={styles["auto-cursor"]} disabled={true}>
-                        {this.props.textBetweenButtons}
-                    </Button>
+                    <If condition={!this.textBoxActive}>
+                        <Then>
+                            <Button key="textBetweenButtons" className={styles["auto-cursor"]} disabled={!this.props.pageNumberEditable}
+                                    onClick={this.handleMiddleButtonClick}>
+                                {this.props.textBetweenButtons}
+                            </Button>
+                        </Then>
+                        <Else>
+                            <FormControl
+                                autoFocus={true}
+                                className={styles["page-number-input"]}
+                                value={this.pageNumberValue}
+                                type="number"
+                                onKeyPress={this.handleCurrentPageSubmit as React.KeyboardEventHandler<any>}
```

Comment:
hard to say without using it but it think it should require submission either by return key or blur? i.e. not just on keypress